eks/mng/scale/scale.go

```diff
@@ -0,0 +1,239 @@
+// Package scale implements EKS cluster scaler tester.
+package scale
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	wait "github.com/aws/aws-k8s-tester/eks/mng/wait"
+	"github.com/aws/aws-k8s-tester/eksconfig"
+	k8s_client "github.com/aws/aws-k8s-tester/pkg/k8s-client"
+	"github.com/aws/aws-k8s-tester/pkg/timeutil"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+	aws_eks "github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/eks/eksiface"
+	"github.com/mitchellh/colorstring"
+	"go.uber.org/zap"
+)
+
+// Scaler defines MNG scaler interface.
+type Scaler interface {
+	// Update starts MNG scaler process, and waits for its completion.
+	// ref. https://docs.aws.amazon.com/cli/latest/reference/eks/update-nodegroup-config.html
+	Scale(mngName string, update *eksconfig.MNGScaleUpdate) error
+}
+
+// Config defines scaler configuration.
+type Config struct {
+	Logger    *zap.Logger
+	Stopc     chan struct{}
+	EKSConfig *eksconfig.Config
+	K8SClient k8s_client.EKS
+	EKSAPI    eksiface.EKSAPI
+}
+
+// New creates a new Scaler.
+func New(cfg Config) Scaler {
+	cfg.Logger.Info("creating tester", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+	return &tester{cfg: cfg}
+}
+
+type tester struct {
+	cfg Config
+}
+
+var reqID = ""
+
+func (ts *tester) Scale(mngName string, update *eksconfig.MNGScaleUpdate) (err error) {
+	cur, ok := ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName]
+	if !ok {
+		ts.cfg.Logger.Warn("MNG not found; failing update", zap.String("mng-name", mngName))
+		return fmt.Errorf("MNGs[%q] not found; failed to update", mngName)
+	}
+	if len(cur.ScaleUpdates) == 0 {
+		ts.cfg.Logger.Info("MNG scaler is not enabled; skipping update", zap.String("mng-name", mngName))
+		return nil
+	}
+	for i, scaleUpdate := range cur.ScaleUpdates {
+		if scaleUpdate.Created {
+			if i == len(cur.ScaleUpdates) - 1 {
+				ts.cfg.Logger.Info("All updates have been applied for this MNG", zap.String("mng-name", mngName))
+				return nil
+			}
+		}
+	}
+
+	if ts.cfg.EKSConfig.LogColor {
+		colorstring.Printf("\n\n[yellow]*********************************[default]\n")
+		colorstring.Printf("[light_green]MNGs[%q].Update[default]\n", mngName)
+	} else {
+		fmt.Printf("\n\n*********************************\n")
+		fmt.Printf("MNGs[%q].Update\n", mngName)
+	}
```

Comment:
This can be simplified as:

```go
fmt.Printf(ts.cfg.EKSConfig.Colorize("\n\n[yellow]*********************************\n"))
fmt.Printf(ts.cfg.EKSConfig.Colorize("[light_green]MNGs[%q].Scale\n"), mngName)
```